src/main/java/com/nike/cerberus/service/AuthenticationService.java

```diff
@@ -402,6 +402,27 @@ private VaultAuthResponse generateToken(final String username, final Set<String>
     }
 
     /**
+     * Build a set of policies that is allowed for the specific IAM principal (e.g. an assumed user, or instance profile)
+     * as well as the generic role ARN (i.e. arn:aws:iam::1111111111:role/example)
+     * @param iamPrincipalArn - The given IAM principal ARN during authentication
+     * @return - List of all policies the given ARN has access to
+     */
+    protected Set<String> buildCompleteSetOfPolicies(final String iamPrincipalArn) {
+
+        final Set<String> allPolicies = buildPolicySet(iamPrincipalArn);
+
+        if (! awsIamRoleArnParser.isRoleArn(iamPrincipalArn)) {
+            logger.debug("Could not find IAM role in principal format, trying 'role' format...");
```

Comment:
This log statement is confusing, because I can authenticate with an instance-profile arn and have 2 SDBs set up in Cerberus one the references only the instance-profile arn and another that references the base role.

Might be better to be something like

    Detected non-role ARN, attempting to collect policies for principals base role

IAM Role ARNs, Instance Profile ARNs, Assumed Role ARNs are all principals so the language around an IAM Role vs a "principal format" is confusing.